src/OrchardCore/OrchardCore.Deployment.Core/Services/DeploymentManager.cs

```diff
@@ -23,9 +23,11 @@ public class DeploymentManager : IDeploymentManager
 
         public async Task ExecuteDeploymentPlanAsync(DeploymentPlan deploymentPlan, DeploymentPlanResult result)
         {
-            foreach (var step in deploymentPlan.DeploymentSteps)
+            var orderedDeploymentSources = new DeploymentSourceSortedList(_deploymentSources);
+
+            foreach (var source in orderedDeploymentSources)
             {
-                foreach (var source in _deploymentSources)
+                foreach (var step in deploymentPlan.DeploymentSteps)
```

Comment:
Not sure a 100% but finally i think that these are the deployment steps that need to be ordered, and this loop would need to be again the main loop as before. I will comment it in the main thread